Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -461,32 +383,29 @@ public class HTTPClient {
     ///     - request: HTTP request to execute.
     ///     - delegate: Delegate to process response parts.
     ///     - eventLoop: NIO Event Loop preference.
+    ///     - context: Baggage context associated with this request
     ///     - deadline: Point in time by which the request must complete.
-    ///     - logger: The logger to use for this request.
     public func execute<Delegate: HTTPClientResponseDelegate>(request: Request,
                                                               delegate: Delegate,
                                                               eventLoop eventLoopPreference: EventLoopPreference,
+                                                              context: LoggingBaggageContextCarrier,
                                                               deadline: NIODeadline? = nil) -> Task<Delegate.Response> {
-        return self.execute(request: request,
-                            delegate: delegate,
-                            eventLoop: eventLoopPreference,
-                            deadline: deadline,
-                            logger: HTTPClient.loggingDisabled)
-    }
+        var span = InstrumentationSystem.tracingInstrument.startSpan(named: request.method.rawValue, context: context, ofKind: .client)
+        span.attributes.http.method = request.method.rawValue
+        span.attributes.http.scheme = request.scheme
+        span.attributes.http.target = request.uri
+        span.attributes.http.host = request.host
+        if let requestContentLength = request.body?.length {
+            span.attributes.http.requestContentLength = requestContentLength
+        }
+
+        // TODO: net.peer.ip / Not required, but recommended
+
+        var request = request
+        InstrumentationSystem.instrument.inject(span.context.baggage, into: &request.headers, using: HTTPHeadersInjector())
+
+        let logger = context.logger.attachingRequestInformation(request, requestID: globalRequestID.add(1))
```

Comment:
@ktoso @slashmo 

this updates an instance of logger,
should it not set requestId in baggage (which then would be used to setup logger)?

Reply:
(copied from https://github.com/swift-aws/aws-sdk-swift-core/pull/348#discussion_r478508462)

how about sth like:

```swift
    private struct RequestMetadata {
        var requestId: Int
        var service: String
        var operation: String
    }
    
    private enum RequestKey: BaggageContextKey {
        typealias Value = RequestMetadata
        var name: String { "AWSRequest" }
    }
 ```

then updating the baggage like:

```swift
context.baggage[RequestKey.self] = RequestMetadata(requestId: Self.globalRequestID.add(1),
    service: serviceConfig.service,
    operation: "signURL")
```

instead of updating the logger directly (current):

```swift
context.logger = context.logger.attachingRequestId(Self.globalRequestID.add(1), operation: "signURL", service: serviceConfig.service)
```